vizro-core/src/vizro/actions/_callback_mapping/_get_action_callback_mapping.py

```diff
@@ -42,5 +44,8 @@ def _get_action_callback_mapping(action_id: ModelID, argument: str) -> Union[Dic
         },
         _on_page_load.__wrapped__: {"inputs": _get_action_callback_inputs, "outputs": _get_action_callback_outputs},
     }
-    action_call = action_callback_mapping.get(action_function, {}).get(argument, {})
-    return action_call if isinstance(action_call, dict) else action_call(action_id=action_id)
+    action_call = action_callback_mapping.get(action_function, {}).get(argument)
+    default_value: Union[List[dcc.Download], Dict[str, Union[Input, State, Output]]] = (
+        [] if argument == "components" else {}
+    )
```

Comment:
It seems like leaky abstraction even more to me that the function returns `{}` if the argument is "components". If some function requires "components", than it expects a list as a response. (empty or not-empty list, but still a list object). 

Let's imagine that there are many places from where the `_get_action_callback_mapping` is called with argument="components". It would mean that this case should be handled on every call. e.g.:

```
components = _get_action_callback_mapping(
    action_id="ACTION_ID",
    argument="components"
) or []  # 'or []' should be places on every _get_action_callback_mapping call.
```

Reply:
I'm not sure I like the dictionary structure in general here, but for now it looks ok I think.